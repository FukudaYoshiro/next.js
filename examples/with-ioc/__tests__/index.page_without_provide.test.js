/* eslint-env jest */

/*
 * Testing pages without @provide decorator as usual
 */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

describe('With Enzyme', () => {
  it('App shows "Menu"', () => {
    const app = shallow(<App />)
    expect(app.find('li a').first().text()).toEqual('Blog: Hello world')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Menu"', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
