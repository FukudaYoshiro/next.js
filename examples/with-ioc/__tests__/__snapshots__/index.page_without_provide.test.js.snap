// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Menu" 1`] = `
<ul>
  <li>
    <a
      href="/blog/hello-world"
      onClick={[Function]}
    >
      Blog: Hello world
    </a>
  </li>
  <li>
    <a
      href="/blog/another-blog-post"
      onClick={[Function]}
    >
      Blog: Another blog post
    </a>
  </li>
  <li>
    <a
      href="/blog/non-existant"
      onClick={[Function]}
    >
      Blog: Not found
    </a>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      About foo bar
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      About foo baz
    </button>
  </li>
</ul>
`;
